src/main/java/com/autotune/common/performanceProfiles/PerformanceProfileInterface/PerfProfileImpl.java

```diff
@@ -84,42 +89,40 @@ public String validateResults(PerformanceProfile performanceProfile, ExperimentR
-                                errorMsg = validateAggFunction(aggrInfoClassAsMap.keySet(), aggrFunctionsObjects);
-                                if (!errorMsg.isBlank()) {
-                                    errorMsg = errorMsg.concat(String.format("for the experiment : %s"
-                                            , experimentResultData.getExperiment_name()));
-                                    break;
-                                }
-                            } catch (IllegalAccessException | InvocationTargetException e) {
-                                throw new RuntimeException(e);
-                            }
-                        } else {
-                            // TODO: check for query and validate against value in kruize object
-                            if (queryList.isEmpty()) {
-                                errorMsg = AnalyzerErrorConstants.AutotuneObjectErrors.QUERY_FUNCTION_MISSING;
-                                break;
-                            } else if (null == funcVar.get("results").getValue()) {
-                                errorMsg = AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_VALUE;
+                }
+                for (HashMap<String, MetricResults> funcVar : containerMetricsMap.values()) {
+                    Map<String, Object> aggrInfoClassAsMap;
+                    if (!aggrFunctionsObjects.isEmpty()) {
+                        try {
+                            aggrInfoClassAsMap = convertObjectToMap(funcVar.get("results").getAggregationInfoResult());
+                            errorMsg = validateAggFunction(aggrInfoClassAsMap.keySet(), aggrFunctionsObjects);
+                            if (!errorMsg.isBlank()) {
+                                errorMsg = errorMsg.concat(String.format("for the experiment : %s"
+                                        , experimentResultData.getExperiment_name()));
                                 break;
                             }
+                        } catch (IllegalAccessException | InvocationTargetException e) {
+                            throw new RuntimeException(e);
                         }
+                    } else {
+                        // TODO: check for query and validate against value in kruize object
+                        if (queryList.isEmpty()) {
+                            errorMsg = AnalyzerErrorConstants.AutotuneObjectErrors.QUERY_FUNCTION_MISSING;
+                            break;
+                        } else if (null == funcVar.get("results").getValue()) {
+                            LOGGER.warn(AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_VALUE);
+                            //TODO: Need to update the below code later
+//                                errorMsg = AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_VALUE;
+//                                break;
```

Comment:
Is it intentional to remove the `break` here? Please remove any commented code if it is not needed. Also is the TODO still relevant. I'd suggest that we need to fix any TODO's related to error messages in the same PR.